### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by margineanmihai**
Refactored Controllers, DB connection, Authentication
Added Routes, Models
Fixed Header bugs

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
+app.set('view engine','ejs');
+app.use(expressLayouts);
+
+// Routes
+const routes = require('./routes/routes');
+app.use(routes);
+
+/* MONGOOSE SETUP */
+const mongoose = require('mongoose');
+mongoose.connect('mongodb+srv://mihai:mihai@cluster0.j9czx.mongodb.net/musicapp?retryWrites=true&w=majority',{
```
Comment (skylock) [inline app.js]: Picking up secrets from the environment is preferred. Committing to the repo should be avoided

**Review by skylock — APPROVED** [review]
Nice work, although smaller commits would make the review process easier.

File: app.js (hunk 1 above)
Comment (margineanmihai) [reply app.js]: Thank you for mentioning that. Fixed it

**Comment by margineanmihai** [comment]
> Nice work, although smaller commits would make the review process easier.

Thank you and you're right. I should have done smaller commits over time.

File: app.js [hunk 2]
```diff
@@ -22,7 +23,7 @@ app.use(routes);
 
 /* MONGOOSE SETUP */
 const mongoose = require("mongoose");
-mongoose.connect("mongodb+srv://mihai:mihai@cluster0.j9czx.mongodb.net/musicapp?retryWrites=true&w=majority",{ 
+mongoose.connect(`${process.env.DB_HOST}://${process.env.DB_USER}:${process.env.DB_PASS}@cluster0.j9czx.mongodb.net/${process.env.DB_NAME}?retryWrites=true&w=majority`,{
```
Comment (skylock) [inline app.js]: Nice work !

**Review by skylock — APPROVED** [review]

File: app.js [hunk 3]
```diff
@@ -1,21 +1,39 @@
...
-app.use(expressLayouts);
-app.use(express.static('./public'));
-app.use(express.json());
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({extended:true}));
-trackController(app);
-loginController(app);
-const passport = require('passport');
+require("dotenv").config();
+const express=require("express");
```
Comment (Radu94) [inline app.js]: spacing is a bit off :) but not the worse thing in the world :)

File: controllers/trackController.js [hunk 4]
```diff
@@ -1,131 +1,87 @@
...
+        if(req.query.username!=null) {
+            res.render("cart",{cartitems:data, title:"Shopping Cart", username:req.query.username});
+        }
+        else {
+            res.render("cart",{cartitems:data, title:"Shopping Cart", username:""});
+        }
+    });       
+};
+
+exports.cartDeleteItem = (req, res) => {
```
Comment (Radu94) [inline controllers/trackController.js]: try to be consistent with your exports :)
as a personal preference i usually export everything i need to on the last line, it makes everything easier to read :)

**Review by Radu94 — APPROVED** [review]

**Pull request merged**